asset_data.py

```diff
@@ -9,31 +9,44 @@ class DataManager:
     def __init__(self):
 
         self.cg_endpoint = "https://api.coingecko.com/api/v3"
-        self.cg_asset_data = {}
+        self.cg_assets = "bitcoin,ethereum,solana,dogecoin"
+        self.cg_assets_list = self.cg_assets.split(",")
+        self.cg_assets_data = {}
         self.cg_global_data = {}
         self.crypto_mcap_perc_total = None
+        self.crypto_total_mcap = None
         # crypto_mcap_perc is a dictionary of top10 assets and percentages of total market cap
         self.cg_header = {
             "x_cg_demo_api_key": os.environ["CG_API_KEY"],
         }
 
-    def update_asset_data(self):
-        """API request to get relevant data for assets selected by users during sign-up"""
-        asset_params = {
-            "ids": "bitcoin,ethereum,solana,dogecoin",
-            "vs_currencies": "usd",
-            "include_market_cap": "true",
-            "include_24hr_change": "true",
-            "include_24h_vol": "true",
-            "include_last_updated_at": "true",
-        }
+    def get_asset_data(self, asset):
+        """Gets user requested data from CoinGecko API and formats it into a dictionary.
+        Also adds commas/punctuation and rounds to two decimal places."""
         asset_response = requests.get(
-            url=f"{self.cg_endpoint}/simple/price",
+            url=f"{self.cg_endpoint}/coins/{asset}",
             headers=self.cg_header,
-            params=asset_params
+            params={"id": asset}
         )
         asset_response.raise_for_status()
-        self.cg_asset_data = asset_response.json()
+        data = asset_response.json()
+        asset_dict = {
+            "ticker": data["symbol"],
+            "price": '${:,.2f}'.format(data["market_data"]["current_price"]["usd"]),
+            "mcap": '${:,.2f}'.format(data["market_data"]["market_cap"]["usd"]),
+            "total_volume": '${:,.2f}'.format(data["market_data"]["total_volume"]["usd"]),
+            "24h_change_usd": '${:,.2f}'.format(data["market_data"]["price_change_24h_in_currency"]["usd"]),
+            "24h_change_percent": f"{round(data["market_data"]["price_change_percentage_24h"], 2)}",
+            "7d_change_percent": f"{round(data["market_data"]["price_change_percentage_7d"], 2)}",
+            "30d_change_percent": f"{round(data["market_data"]["price_change_percentage_30d"], 2)}",
+        }

```

Comment:
This is totally fine but this may be more readable 


```suggestion
      ticker = data["symbol"]
      price = '${:,.2f}'.format(data["market_data"]["current_price"]["usd"])
      mcap = '${:,.2f}'.format(data["market_data"]["market_cap"]["usd"])
      // do the same for each variable

        asset_dict = {
            "ticker": ticker,
            "price": price,
            "mcap": mcap,
            "total_volume": '${:,.2f}'.format(data["market_data"]["total_volume"]["usd"]),
            "24h_change_usd": '${:,.2f}'.format(data["market_data"]["price_change_24h_in_currency"]["usd"]),
            "24h_change_percent": f"{round(data["market_data"]["price_change_percentage_24h"], 2)}",
            "7d_change_percent": f"{round(data["market_data"]["price_change_percentage_7d"], 2)}",
            "30d_change_percent": f"{round(data["market_data"]["price_change_percentage_30d"], 2)}",
        }

```